### DISCUSSION TITLE
How to import dependencies that contain alised imports in nuxt3 modules

### DISCUSSION THREAD

**Discussion by shtse8** [nuxt/nuxt#17312] [Questions]
I am making my own module and need to import some libraries relative to the project root.

```
 ERROR  Cannot start nuxt:  Cannot find module '~/lib/prisma'                                                                                                                                                                                             01:19:07
Require stack:
- D:\nuxtjs\modules\hooks.ts
```

**Comment by danielroe** [comment]
You should inject a plugin. There is a key difference between _build-time_ and _runtime_ code. Aliases are supported in runtime code (like plugins, or other templates you might inject), but not in _build-time_ code. 

The best way to start is to have a look at how a similar module does things. Check out modules.nuxtjs.org - and especially [this module guide](https://v3.nuxtjs.org/guide/going-further/modules).

**Reply by shtse8** [reply]
thanks.
so how to make it work if I need to use it in `builder:watch` event hook? 
there is no this hook in plugin.

I need to close the database connection before restart to prevent crash.
```
error Command failed with exit code 3221225477.
```

**Reply by danielroe** [reply]
I think probably we would need a lot more information from you about what you are trying and what your code looks like. But if you need access in your build-time module, then don't use an alias; you can just manually come up with the path:

```js
const prismaPath = join(nuxt.options.srcDir, 'lib/prisma')
const prisma = await import(prismicPath')
// do magic
```

There's even a useful kit utility `resolveAlias` that will do this for you:

```js
const prismaPath = resolveAlias('~/lib/prisma')
```

**Comment by shtse8** [comment]
```ts
import { defineNuxtModule } from '@nuxt/kit'
import { usePrisma } from ''~/lib/prisma'

export default defineNuxtModule({
  setup(options, nuxt) {
    nuxt.hook('builder:watch', (event, path) => {
      const prisma = usePrisma()
      prisma.$disconnect()
    })
  }
}
```

I can't import my library using `~` alias, but it works fine with absolute path.
It's hard to change it to absolute path because all nested imports are required to change.
Any better method to auto resolve all nested alias?
